### DISCUSSION TITLE
Cascading dropdown - how to use

### DISCUSSION THREAD

**Discussion by SigalZ** [livewire/livewire#4525] [1. Help]
Following this tutorial of creating a dynamic dropdown: https://www.itsolutionstuff.com/post/laravel-livewire-dependant-dropdown-exampleexample.html

I have this dropdown component:

```
public $suppliers;
    public $beans;
    public $selectedSupplier = NULL;

    public function mount()
    {
        $this->suppliers = Supplier::whereHas('greenBeans')->get();
        
        $this->beans = collect();
    }

    public function render()
    {
        return view('livewire.admin.supplier-beans-dropdown')->layout('layouts.admin.livewire');
    }  

    public function updatedSelectedSupplier($supplier)
    {
        if (!is_null($supplier)) {
            $supplier = Supplier::find($supplier);
            $this->beans = $supplier->greenBeans;
        }
    }    

```
I have an other component that needs to call this dropdown. So in this parent component I have this var: $selectedSupplier and in the view I call the dropdown component:

`@livewire('admin.supplier-beans-dropdown')`

When I select a supplier and submit the form, the selectSupplier is null.

So how do I use this dropdown in different components?
How do I send the selected value to the parent component?
If I need this dropdown in 5 different forms, do I need to rewrite this code in each component?

**Comment by joshhanley** [comment]
@SigalZ I'd strongly recommend not splitting it into Livewire components as the only way the can communicate is through sending events between them.

You'd be better off just creating one component with all the dropdowns in the one component. I'd use Laravel's blade components to split the view up if it becomes too long.

Hope this helps!

**Reply by SigalZ** [reply]
Thank you joshhanley.
I just couldn't understand what is the point of all these tutorial of cascading dropdowns if you can't use the component itself in other components.

So I ended up creating the supplier/green beans dropdown in each of the livewire components I needed it in.

Is that what you mean?

**Reply by joshhanley** [reply]
@SigalZ yeah some of the tutorials out there aren't the best as I think they try to write it too much like Vue but Livewire doesn't work in the same way.

Yep that's what I meant 🙂
